climateconnect/climateconnect
# Redirect logged in users from '/' to '/browse' without memory leak

## Context

We recently removed our `getServerSideProps` function from `index.js`. All this function was handling was redirecting users from `/` to `/browse` if they were already logged in because the `/` page is just a landing page for new users. However this function seemed to be called over and over even after the redirect was done and was therefore causing a server-side memory leak.
We removed the redirect in a prior update to make sure the memory leak was caused by this which turned out to be the case. 

## Changes
- The described redirect is now happening through the "redirects" function in next.config.js. This seemed to be the easiest and most performant solution
- If the `token` cookie is set you will be redirected to `/browse`. If it's not set you will just go to `/`

## Test plan

- potentially create a production build for testing by using `yarn build` and then running it with `yarn start`
- Try to access `localhost:3000` while not logged in. You should not be redirected
- Try to access `localhost:3000`while logged in. You should be redirected to `/[your_locale]/browse`
- Try to change the language of the website and visit `localhost:3000` again while logged in. The redirect should point to the `browse` page in the correct language

## Before landing

1. PR has meaningful title
1. `yarn lint` passes (frontend)
1. `yarn format` passes (frontend)
1. `make format` passes (backend)
